Meta/Lagom/CMakeLists.txt

```diff
@@ -51,281 +79,459 @@ set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} ${LINKER_FLAGS}")
+#        in combination with the system, we generate two Debug.h files. One in
+#        Build/AK/Debug.h and the other in Build/Meta/Lagom/AK/Debug.h.
+configure_file(../../AK/Debug.h.in AK/Debug.h @ONLY)
+configure_file(../../Kernel/Debug.h.in Kernel/Debug.h @ONLY)
+
 include_directories(../../)
 include_directories(../../Userland/)
 include_directories(../../Userland/Libraries/)
 include_directories(${CMAKE_BINARY_DIR})
 include_directories(${CMAKE_CURRENT_BINARY_DIR})
 
-if (BUILD_LAGOM AND NOT ENABLE_OSS_FUZZ AND NOT ENABLE_FUZZER_SANITIZER)
-    if (CMAKE_SOURCE_DIR MATCHES ".*/Lagom")
-        set(write_if_different ${CMAKE_SOURCE_DIR}/../write-only-on-difference.sh)
-        add_subdirectory(../../Userland/Libraries/LibUnicode/CodeGenerators ${CMAKE_CURRENT_BINARY_DIR}/LibUnicode/CodeGenerators)
-    endif()
+# install rules, think about moving to its own helper cmake file
+# Don't install Lagom libs into the target Root/
+# FIXME: Remove this check for 4594
+if (CMAKE_SOURCE_DIR MATCHES ".*/Lagom")
+    include(CMakePackageConfigHelpers)
+    include(GNUInstallDirs)
+
+    set(package Lagom)
+
+    write_basic_package_version_file(
+        "${package}ConfigVersion.cmake"
+        COMPATIBILITY SameMajorVersion
+    )
+
+    # Allow package maintainers to freely override the path for the configs
+    set(Lagom_INSTALL_CMAKEDIR "${CMAKE_INSTALL_DATADIR}/${package}"
+        CACHE PATH "CMake package config location relative to the install prefix")
+    mark_as_advanced(Lagom_INSTALL_CMAKEDIR)
+
+    install(
+        FILES ${SERENITY_PROJECT_ROOT}/Meta/CMake/lagom-install-config.cmake
+        DESTINATION "${Lagom_INSTALL_CMAKEDIR}"
+        RENAME "${package}Config.cmake"
+        COMPONENT Lagom_Development
+    )
+
+    install(
+        FILES "${PROJECT_BINARY_DIR}/${package}ConfigVersion.cmake"
+        DESTINATION "${Lagom_INSTALL_CMAKEDIR}"
+        COMPONENT Lagom_Development
+    )
+
+    install(
+        EXPORT LagomTargets
+        NAMESPACE Lagom::
+        DESTINATION "${Lagom_INSTALL_CMAKEDIR}"
+        COMPONENT Lagom_Development
+    )
+
+    # Manually install AK
+    install(DIRECTORY "${SERENITY_PROJECT_ROOT}/AK"
+        COMPONENT Lagom_Development
+        DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
+        FILES_MATCHING PATTERN "*.h")
 
-    include(../../Userland/Libraries/LibUnicode/unicode_data.cmake)
 endif()
 
+function(lagom_lib library fs_name)
+    cmake_parse_arguments(LAGOM_LIBRARY "" "" "SOURCES;LIBS" ${ARGN})
+    # FIXME: Consider whether to care about -DBUILD_SHARED_LIBS=OFF
+    # Possibly a cmake presets value?
+    set(target_name "Lagom${library}")
+    add_library(${target_name} SHARED ${LAGOM_LIBRARY_SOURCES})
+    # alias for pretty exports
+    add_library(Lagom::${target_name} ALIAS ${target_name})
+
+    set_target_properties(
+        ${target_name} PROPERTIES
+        VERSION "${PROJECT_VERSION}"
+        SOVERSION "${PROJECT_VERSION_MAJOR}"
+        EXPORT_NAME ${target_name}
```

Comment:
[EXPORT_NAME](https://cmake.org/cmake/help/latest/prop_tgt/EXPORT_NAME.html) is already set to the target's name. Maybe you meant to use `library` here?

Taking `JS` as an example, `target_name` will be `LagomJS` and the export will be `Lagom::LagomJS`. Is this intended? `Lagom::JS` would make more sense.

Reply:
Oh! that makes a lot of sense. I've gone ahead and changed to `${library} for the EXPORT_NAME and for the local Lagom::${library} alias. a few lines up. That'll make the `target_link_libraries` in https://github.com/linusg/libjs-test262/pull/30  look better for sure. 

Although.. I'm sure there's a more canonical CMake-y way to approach that PR as well. The main confusing thing is that we want to be able to run the libjs-test262 test runner locally with a local build directory in the user's `serenity` checkout easily, but also run it more or less painlessly in the CI for the main serenity repo with each master commit. ... And to not have to build all 600 files, when only ~300 are required to build `Lagom::JS`. Which it doesn't seem like ExternalProject is the correct tool for :/

Reply:
Never mind on that secondary comment, I found FETCHCONTENT_SOURCE_DIR_LAGOM and it seems to do the trick nicely.